pkg/router/auth.go

```diff
@@ -0,0 +1,162 @@
+package router
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+
+	fv1 "github.com/fission/fission/pkg/apis/core/v1"
+	config "github.com/fission/fission/pkg/featureconfig"
+)
+
+var (
+	malformedToken = errors.New("Unauthorized: malformed token")
+	expiredToken   = errors.New("Unauthorized: token is either expired or not active yet")
+	invalidCreds   = errors.New("Unauthorized: invalid username or password")
+	authPath       = "/auth/login"
+)
+
+func authMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != authPath && r.URL.Path != "/router-healthz" {
```

Comment:
Please negate this condition so that we dont have to indent rest of the block.

Also we can define move this code to a method.

```
func checkAuthToken(r *http.Request) error 
```

We can call this function and if error is not nil, we serve the request other, otherwise we return http.StatusUnauthorized with respective error.